src/snowflake/snowpark/modin/plugin/_internal/aggregation_utils.py

```diff
@@ -414,34 +427,125 @@ def _columns_coalescing_sum(*cols: SnowparkColumn) -> Callable:
 
 
 # Map between the pandas input aggregation function (str or numpy function) and
-# the corresponding aggregation function for axis=1 when skipna=True. The returned aggregation
-# function may either  be a builtin aggregation function, or a function taking in *arg columns
-# that then calls the appropriate builtin aggregations.
-SNOWFLAKE_COLUMNS_AGG_FUNC_MAP: dict[Union[str, Callable], Callable] = {
-    "count": _columns_count,
-    "sum": _columns_coalescing_sum,
-    np.sum: _columns_coalescing_sum,
-    "min": _columns_coalescing_min,
-    "max": _columns_coalescing_max,
-    "idxmax": _columns_coalescing_idxmax_idxmin_helper,
-    "idxmin": _columns_coalescing_idxmax_idxmin_helper,
-    np.min: _columns_coalescing_min,
-    np.max: _columns_coalescing_max,
-}
-
-# These functions are called instead if skipna=False
-SNOWFLAKE_COLUMNS_KEEPNA_AGG_FUNC_MAP: dict[Union[str, Callable], Callable] = {
-    "min": least,
-    "max": greatest,
-    "idxmax": _columns_coalescing_idxmax_idxmin_helper,
-    "idxmin": _columns_coalescing_idxmax_idxmin_helper,
-    # IMPORTANT: count and sum use python builtin sum to invoke __add__ on each column rather than Snowpark
-    # sum_, since Snowpark sum_ gets the sum of all rows within a single column.
-    "sum": lambda *cols: sum(cols),
-    np.sum: lambda *cols: sum(cols),
-    np.min: least,
-    np.max: greatest,
-}
+# _SnowparkPandasAggregation representing information about applying the
+# aggregation in Snowpark pandas.
+_PANDAS_AGGREGATION_TO_SNOWPARK_PANDAS_AGGREGATION: MappingProxyType[
+    AggFuncTypeBase, _SnowparkPandasAggregation
+] = MappingProxyType(
+    {
+        "count": _SnowparkPandasAggregation(
+            axis_0_aggregation=count,
+            axis_1_aggregation_skipna=_columns_count,
+            preserves_snowpark_pandas_types=False,
+        ),
+        **{
+            k: _SnowparkPandasAggregation(
+                axis_0_aggregation=mean,
+                preserves_snowpark_pandas_types=True,
+            )
+            for k in python_cast(tuple[AggFuncTypeBase], ("mean", np.mean))
```

Comment:
to further improve the readability, can we have a method called create_pandas_to_snowpark_pandas_aggression_map. and this can be
```
create_pandas_to_snowpark_pandas_aggression_map("count", _SnowparkPandasAggregation(
            axis_0_aggregation=count,
            axis_1_aggregation_skipna=_columns_count,
            preserves_snowpark_pandas_types=False,
        ))
**create_pandas_to_snowpark_pandas_aggression_map(("mean", np.mean), _SnowparkPandasAggregation(
                axis_0_aggregation=mean,
                preserves_snowpark_pandas_types=True,
            ))
```

Reply:
Done.